### DISCUSSION TITLE
NuGet path conventions

### DISCUSSION THREAD

**Discussion by martasp** [NuGet/Home#11608] [General]
Why package name ` microsoft.codeanalysis.common` folder is not the same as assembly dll name for `Microsoft.CodeAnalysis.dll`?
Most NuGet packages follow a convention where the package name is the same as the assembly dll name.
Maybe there is no such convention?

Here some path examples : 
".nuget\\packages\\cocona\\2.0.3\\lib\\netstandard2.0\\Cocona.dll"✅
".nuget\\packages\\cocona.core\\2.0.3\\lib\\netstandard2.0\\Cocona.Core.dll"✅
".nuget\\packages\\diffengine\\6.4.9\\lib\\netstandard2.0\\DiffEngine.dll"✅
".nuget\\packages\\emptyfiles\\2.3.3\\lib\\netstandard2.0\\EmptyFiles.dll"✅
".nuget\\packages\\microsoft.bcl.asyncinterfaces\\5.0.0\\lib\\netstandard2.0\\Microsoft.Bcl.AsyncInterfaces.dll"✅
".nuget\\packages\\microsoft.codeanalysis.common\\4.1.0\\lib\\netstandard2.0\\Microsoft.CodeAnalysis.dll"❌
".nuget\\packages\\microsoft.codeanalysis.csharp\\4.1.0\\lib\\netstandard2.0\\Microsoft.CodeAnalysis.CSharp.dll"✅

**Comment by PathogenDavid** [comment]
> Why package name ` microsoft.codeanalysis.common` folder is not the same as assembly dll name for `Microsoft.CodeAnalysis.dll`?

Because while the package ID and the assembly name are typically the same, they don't have to be.

In the case of Roslyn, referencing the assembly `Microsoft.CodeAnalysis.dll` is not something you usually want to do directly. So it lives in a different package and the `Microsoft.CodeAnalysis` package is a meta package which references all the assemblies a typical consumer of Roslyn might need.
